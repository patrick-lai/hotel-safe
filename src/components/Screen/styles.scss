@import '../../theme/default.scss';

.screen {
  border: 8px solid black;
  padding: 1em;
  background-color: $color-screen;
  border-radius: 20px;
  height: 50%;
  box-shadow: inset 0 0 20px #333;
  font-family: monospace;
  font-size: 400%;
  display: flex;
  justify-content: center;
  align-items: center;
  letter-spacing: 10px;
  transition: all 250ms ease;

  &.-error {
    color: $color-fail;
  }

  &.-success {
    color: $color-success;
  }
}
