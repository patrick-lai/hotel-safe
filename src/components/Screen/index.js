/***
 * Screen displaying the lock's status
 */

import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Screen = ({ display, status }) => {
  // Build the full classname
  const classNames = ['screen'];
  switch (status) {
    case 'INVALID':
      classNames.push('-error');
      break;
    default:
    // Do nothing
  }

  return (
    <div data-test-id="screen" className={classNames.join(' ')}>
      {display}
    </div>
  );
};

Screen.propTypes = {
  display: PropTypes.string,
  status: PropTypes.string
};

export default Screen;
