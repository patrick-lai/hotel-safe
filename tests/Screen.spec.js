/***
 * Screen
 */

import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Screen from '../src/components/Screen';

Enzyme.configure({ adapter: new Adapter() });

describe('Screen component', () => {
  test('should be able to render a string', () => {
    const wrapper = shallow(<Screen display="sample" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should have correct classes on different states', () => {
    const none = shallow(<Screen display="none" />);
    const success = shallow(<Screen display="success" status="SUCCESS" />);
    const error = shallow(<Screen display="error" status="ERROR" />);
    expect(none.html()).toMatchSnapshot();
    expect(success.html()).toMatchSnapshot();
    expect(error.html()).toMatchSnapshot();
  });
});
