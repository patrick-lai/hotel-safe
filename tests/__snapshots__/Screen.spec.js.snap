// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen component should be able to render a string 1`] = `"<div data-test-id=\\"screen\\" class=\\"screen\\">sample</div>"`;

exports[`Screen component should have correct classes on different states 1`] = `"<div data-test-id=\\"screen\\" class=\\"screen\\">none</div>"`;

exports[`Screen component should have correct classes on different states 2`] = `"<div data-test-id=\\"screen\\" class=\\"screen -success\\">success</div>"`;

exports[`Screen component should have correct classes on different states 3`] = `"<div data-test-id=\\"screen\\" class=\\"screen -error\\">error</div>"`;
